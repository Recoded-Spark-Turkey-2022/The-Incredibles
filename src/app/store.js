import { configureStore } from '@reduxjs/toolkit';
import usersReducer from '../features/users/usersSlice';

export const store = configureStore({
  reducer: {
    users: usersReducer,
  },
});
