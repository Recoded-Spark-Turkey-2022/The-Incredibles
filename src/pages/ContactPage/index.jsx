import React from 'react';
import largeScreen from '../../assets/pics/contactPage/largeScreen.svg';
import smallScreen from '../../assets/pics/contactPage/smallScreen.svg';
import faceIcon from '../../assets/pics/contactPage/faceIcon.svg';
import instaIcon from '../../assets/pics/contactPage/instaIcon.svg';
import twitterIcon from '../../assets/pics/contactPage/twitterIcon.svg';
import { useDispatch } from 'react-redux';
import { writeContact } from '../../features/contact/contactSlice';
import { useState } from 'react';
import Container from '../../components/UI/Container';

function ContactPage() {
  const dispatch = useDispatch();
  const [formData, setFormData] = useState({
    email: '',
    message: '',
  });
  const handleChange = (event) => {
    setFormData({ ...formData, [event.target.id]: event.target.value });
  };
  const handleSubmit = (event) => {
    event.preventDefault();
    dispatch(writeContact(formData));
  };
  return (
    <Container>
      <div className=" flex flex-col md:flex-row-reverse justify-center items-center ">
        <div className="basis-1/2 m-auto">
          <img src={largeScreen} className="m-auto hidden md:flex" />
          <img src={smallScreen} className="m-auto md:hidden" />
        </div>
        <div className=" basis-1/2  text-center md:text-left m-auto md:max-w-xl">
          <h1 className="pb-10 font-bold text-3xl text-cyan-600 md:text-6xl">
            Get in Touch
          </h1>
          <p className="text-zinc-400	font-light pb-2 ">
            But Brooke Chaffin and Catherine Connors are looking to change that
            with the introduction.
          </p>
          <form
            onSubmit={handleSubmit}
            className=" pl-1 flex flex-col gap-2 justify-center md:justify-start "
          >
            <label className="text-zinc-400	font-light text-left ">Email</label>
            <input
              id="email"
              type="email"
              placeholder="hello@gmail.com"
              className="border border-solid"
              onChange={handleChange}
            />
            <textarea
              id="message"
              placeholder="Message"
              rows={8}
              className="border border-solid"
              onChange={handleChange}
            ></textarea>
            <div>
              <input
                type="submit"
                value="Send"
                className="m-auto  h-10 px-10 py-2.5 bg-cyan-600 text-white font-medium text-l leading-tight 
             rounded-full shadow-md
             ease-in duration-300 hover:bg-purple-700 hover:shadow-md  hover:scale-110"
              />
            </div>

            <div className="flex flex-row py-9  justify-center md:justify-start ">
              <img src={twitterIcon} alt="" className="p-2 m-auto md:m-0" />
              <img src={instaIcon} alt="" className="p-2 m-auto md:m-0" />
              <img src={faceIcon} alt="" className="p-2 m-auto md:m-0" />
            </div>
          </form>
        </div>
      </div>
    </Container>
  );
}

export default ContactPage;
