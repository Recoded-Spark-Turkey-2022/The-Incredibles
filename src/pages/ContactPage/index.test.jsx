import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import ContactPage from '.';
import { store } from './../../app/store';
import { Provider } from 'react-redux';

describe('ContactPage', () =>
  it('renders correctly', () => {
    const domTree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <ContactPage />
          </Provider>
        </Router>
      )
      .toJSON();
    expect(domTree).toMatchSnapshot();
  }));
