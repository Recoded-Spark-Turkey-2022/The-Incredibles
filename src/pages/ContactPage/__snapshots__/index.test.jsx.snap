// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage renders correctly 1`] = `
<div
  className="container max-w-screen-xl m-auto p-10 max-md:px-5"
>
  <div
    className=" flex flex-col md:flex-row-reverse justify-center items-center "
  >
    <div
      className="basis-1/2 m-auto"
    >
      <img
        className="m-auto hidden md:flex"
        src="largeScreen.svg"
      />
      <img
        className="m-auto md:hidden"
        src="smallScreen.svg"
      />
    </div>
    <div
      className=" basis-1/2  text-center md:text-left m-auto md:max-w-xl"
    >
      <h1
        className="pb-10 font-bold text-3xl text-cyan-600 md:text-6xl"
      >
        contactpage.get
      </h1>
      <p
        className="text-zinc-400	font-light pb-2 "
      >
        contactpage.p
      </p>
      <form
        className=" pl-1 flex flex-col gap-2 justify-center md:justify-start "
        onSubmit={[Function]}
      >
        <label
          className="text-zinc-400	font-light text-left "
        >
          contactpage.email
        </label>
        <input
          className="border border-solid"
          id="email"
          onChange={[Function]}
          placeholder="contactpage.emailplace"
          type="email"
        />
        <textarea
          className="border border-solid"
          id="message"
          onChange={[Function]}
          placeholder="contactpage.massageplace"
          rows={8}
        />
        <div>
          <input
            className="m-auto  h-10 px-10 py-2.5 bg-cyan-600 text-white font-medium text-l leading-tight 
 rounded-full shadow-md
 ease-in duration-300 hover:bg-purple-700 hover:shadow-md  hover:scale-110"
            type="submit"
            value="contactpage.send"
          />
        </div>
        <div
          className="flex flex-row py-9  justify-center md:justify-start "
        >
          <img
            alt=""
            className="p-2 m-auto md:m-0"
            src="twitterIcon.svg"
          />
          <img
            alt=""
            className="p-2 m-auto md:m-0"
            src="instaIcon.svg"
          />
          <img
            alt=""
            className="p-2 m-auto md:m-0"
            src="faceIcon.svg"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
